GCRCatalogs/reference_catalog.py

```diff
@@ -0,0 +1,79 @@
+"""
+Reference Catalog Reader
+"""
+import os
+import numpy as np
+from GCR import BaseGenericCatalog
+
+__all__ = ['ReferenceCatalogReader']
+
+class ReferenceCatalogReader(BaseGenericCatalog):
+    """
+    Reference Catalog Reader
+
+    Parameters
+    ----------
+    filename : str
+    nlines : None or int, optional (default: 10000)
+        how many lines to read at once
+    max_chunks : None or int, optional (default: None)
+        how many chunks to read.
+        Set to 1 if you just want to test the reader.
+        Set to None to read all chunks.
+    """
+
+    def _subclass_init(self, **kwargs):
+        self._filename = kwargs['filename']
+        if not os.path.isfile(self._filename):
+            raise ValueError('File {} not found'.format(self._filename))
+
+        self._nlines = kwargs.get('nlines', 10000)
+        self._nlines = None if self._nlines is None else int(self._nlines)
+
+        self._max_chunks = kwargs.get('max_chunks')
+        self._max_chunks = None if self._max_chunks is None else int(self._max_chunks)
+
+        self._quantity_modifiers = {
+            'ra' : 'raJ2000',
+            'dec' : 'decJ2000',
+            'sigma_ra' : 'sigma_raJ2000',
+            'sigma_dec' : 'sigma_decJ2000',
+            'ra_smeared' : 'raJ2000_smeared',
+            'dec_smeared' : 'decJ2000_smeared',
+        }
+
+        self._header_line_number = 0
+        self._data_dtype = None
+
+
+    def _iter_native_dataset(self, native_filters=None):
+        if native_filters is not None:
+            raise ValueError('`native_filter` not supported!')
+
+        with open(self._filename, 'rb') as f:
+            for _ in range(self._header_line_number):
+                next(f, None)
+
+            chunk_count = 0
+            while self._max_chunks is None or chunk_count < self._max_chunks:
+                data = np.genfromtxt(f, self._data_dtype, delimiter=',', max_rows=self._nlines)
+                if len(data) == 0:
+                    break
+                yield data.__getitem__
+                chunk_count += 1
+
+
+    def _generate_native_quantity_list(self):
+        line = None
+        with open(self._filename, 'r') as f:
+            for i, line in enumerate(f):
+                if len(line) > 2 and line[0] == '#'  and line[1] not in (' ', '\n'):
```

Comment:
I'm worried that this is a little fragile against someone making the decision that they want the header line to look like `# col_name1, col_name2...` as opposed to `#col_name1, col_name2, ...` (as it does now).

Would it be safer to just choose the last line that starts with `#`?

Or look for a line that contains `uniqueId`, `raJ2000`, and `decJ2000` (all known column names)?

I'm just being paranoid.  I don't actually have strong feelings about this.